api4/team.go

```diff
@@ -1357,6 +1357,29 @@ func importTeam(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(model.MapToJSON(data)))
 }
 
+func validateUserPermissionsOnChannels(c *Context, e model.EmailInvite) {
+	var allowedChannelIds []string
+	channels := e.GetChannels()
+
+	for _, channelId := range channels {
+		channel, err := c.App.GetChannel(c.AppContext, channelId)
+		if err != nil {
+			mlog.Info("Invite users to team - couldn't get channel " + channelId)
+			continue
+		}
+		canManagePrivateChannels := c.App.SessionHasPermissionToChannel(c.AppContext, *c.AppContext.Session(), channelId, model.PermissionManagePrivateChannelMembers)
+		canManagePublicChannels := c.App.SessionHasPermissionToChannel(c.AppContext, *c.AppContext.Session(), channelId, model.PermissionManagePublicChannelMembers)
```

Comment:
It's not as efficient as possible to be invoking `SessionHasPermissionToChannel` twice per channel. Rather we could conditionally decide which permission to check and make one invocation per channel.